dll/win32/shell32/folders/CDrivesFolder.cpp

```diff
@@ -573,6 +573,7 @@ HRESULT WINAPI CDrivesFolder::ParseDisplayName(HWND hwndOwner, LPBC pbc, LPOLEST
     HRESULT hr = E_INVALIDARG;
     LPCWSTR szNext = NULL;
     LPITEMIDLIST pidlTemp = NULL;
+    CComHeapPtr<WCHAR> volumePathName;
```

Comment:
`WCHAR volumePathName[MAX_PATH];`

Reply:
Sounds good!  I've forced pushed that change.

Reply:
What benefits does an array of fixed size of MAX_PATH? Wasn't a pointer just enough?
I may be wrong but nowadays certain paths can exceed above the size of a MAX_PATH. Dynamic buffer allocation is more concise in this matter.

Reply:
I'm happy either way, although the original allocation I had was for MAX_PATH as well.  In this case, volumePathName only stores the path to the volume, and ignores the remainder of the path.  So this code will only fail if the volume name is larger than MAX_PATH (also since we pass MAX_PATH into GetVolumePathNameW it should return an error code if the buffer is not large enough).  The safest would be to get the length of the volume name some how, but I'm not sure if there's a good way to do that.

A future refactor should be to ensure we support up to 32,768 character paths when using the Unicode functions, since I think the Unicode versions support much longer path names.